pdns/syncres.cc

```diff
@@ -388,6 +388,26 @@ uint64_t SyncRes::doDumpNSSpeeds(int fd)
   return count;
 }
 
+uint64_t SyncRes::doDumpThrottleMap(int fd)
+{
+  FILE* fp=fdopen(dup(fd), "w");
+  if(!fp)
+    return 0;
+  fprintf(fp, "; throttle map dump follows\n");
+  fprintf(fp, "; remote IP, DNSName, QType\n");
+  uint64_t count=0;
+
+  for(const auto& i : t_sstorage.throttle.getThrottleTuples())
+  {
+    count++;
+    // remote IP, dns name, qtype
+    fprintf(fp, "%s %s %d", i.get<0>().toLogString().c_str(), i.get<1>().toString().c_str(), i.get<2>());
```

Comment:
It would be nice to have the content of the `entry` as well, the `TTD` and the `count` fields.